components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java

```diff
@@ -192,6 +192,35 @@ public Secret replaceSecret(String secretTypeName, Secret secret) throws SecretM
         return secret;
     }
 
+    @Override
+    public Secret updateSecretValue(String secretTypeName, String name, String value) throws SecretManagementException {
+
+        validateSecretManagerEnabled();
+        Secret secret, updatedSecret;
+        secret = getSecret(secretTypeName, name);
```

Comment:
Can we rethink about do we really need to pull up the complete secret object?
Can't we get the secret ID and continue?

Reply:
After updating the secret, it is removed from the cache. For that needed secret.
https://github.com/wso2/carbon-identity-framework/pull/3725/files#diff-1bc28b6a49428fd9f1e769c37bb6996f3ac9c76509290e883c87cacb9b161e72R140

Reply:
In response we send secret as a whole with new updated time and modified values. So if we send those we have to get the secret at somepoint.